lib/views/status-bar-view.coffee

```diff
@@ -187,6 +204,27 @@ class StatusBarView extends View
                           .text('Success!')
                           .show()
 
+  updateBuildTarget: ->
+    @main.getBuildTargets().then (targets) =>
+      currentBuildTarget = @main.getCurrentBuildTarget()
+      # Clear build target if it doesn't exist for current platform
+      targetExists = targets.reduce (acc, val) =>
+        acc || val.version == currentBuildTarget
+      , false
+      currentBuildTarget = undefined if not targetExists
+
+      if !currentBuildTarget
+        latestVersion = targets.reduce (acc, val) =>
```

Comment:
fetching the latest release/pre-release, and checking if a given version exists for a platform seem like operations that should exist in particle API js, or at least eternally to particle-dev.